src/instructlab/sdg/utils/chunkers.py

```diff
@@ -68,185 +69,211 @@ def resolve_ocr_options() -> OcrOptions:
+#                 paths to documents.
+#         """
+#         doc_dict = defaultdict(list)
+#         for doc, path in zip(documents, filepaths):
+#             if path.suffix == ".md":
+#                 # append doc contents
+#                 doc_dict[SupportedFileTypes.MD].append((doc, path))
+#             elif path.suffix == ".pdf":
+#                 # append doc paths
+#                 doc_dict[SupportedFileTypes.PDF].append((doc, path))
+#             else:
+#                 raise ValueError(
+#                     f"Received document of type .{path.suffix}, which is not a supported filetype"
+#                 )
+#         return doc_dict
+
+
+# class TextSplitChunker(ChunkerBase):
+#     def __init__(
+#         self,
+#         document_contents: List | str,
+#         server_ctx_size: int,
+#         chunk_word_count: int,
+#         output_dir: Path,
+#     ):
+#         self.document_contents = document_contents
+#         self.server_ctx_size = server_ctx_size
+#         self.chunk_word_count = chunk_word_count
+#         self.output_dir = output_dir
+
+#     def chunk_documents(self) -> List:
+#         """Naively chunk markdown documents based on the word count provided by the user.
+#         Returns:
+#             List[str]: List of chunked documents.
+#         """
+#         num_tokens_per_doc = _num_tokens_from_words(self.chunk_word_count)
+#         if num_tokens_per_doc > int(self.server_ctx_size - 1024):
+#             raise ValueError(
+#                 "Error: {}".format(
+#                     str(
+#                         f"Given word count ({self.chunk_word_count}) per doc will exceed the server context window size ({self.server_ctx_size})"
+#                     )
+#                 )
+#             )
+#         if self.document_contents == []:
+#             return []
+
+#         chunk_size = _num_chars_from_tokens(num_tokens_per_doc)
+#         return chunk_markdowns(self.document_contents, chunk_size)
+
+
+class DocumentChunker():  # pylint: disable=too-many-instance-attributes
+
+    # def __new__(
+    #     cls,
+    #     leaf_node,
+    #     taxonomy_path,
+    #     output_dir: Path,
+    #     server_ctx_size=4096,
+    #     chunk_word_count=1024,
+    #     tokenizer_model_name: Optional[str] = None,
+    #     docling_model_path: Optional[str] = None,
+    # ):
 
-class TextSplitChunker(ChunkerBase):
     def __init__(
         self,
-        document_contents: List | str,
-        server_ctx_size: int,
-        chunk_word_count: int,
+        document_paths: List[Path],
         output_dir: Path,
+        tokenizer_model_name: str,
+        docling_model_path: Optional[Path] = None,
+        server_ctx_size: int = 4096,
+        chunk_word_count: int = 1024,
     ):
-        self.document_contents = document_contents
-        self.server_ctx_size = server_ctx_size
-        self.chunk_word_count = chunk_word_count
-        self.output_dir = output_dir
+        if len(document_paths) == 0:
```

Comment:
Just an FYI, in Python you can do `if not document_path` and it will perform both `if document_path is None or len(document_paths) == 0`